pkg/clusteragent/clusterchecks/dispatcher_nodes.go

```diff
@@ -122,3 +122,26 @@ func (d *dispatcher) expireNodes() {
 		log.Warn("No nodes reporting, cluster checks will not run")
 	}
 }
+
+// updateRunnersStats collects stats from the registred
+// Cluster Level Check runners and updates the stats cache
+func (d *dispatcher) updateRunnersStats() {
+	if d.clcRunnersClient == nil {
+		log.Debug("Couldn't instantiate a CLC Runner client")
+		return
+	}
+
+	d.store.Lock()
+	defer d.store.Unlock()
+	for name, node := range d.store.nodes {
+		node.Lock()
+		stats, err := d.clcRunnersClient.GetRunnerStats(node.clientIP)
```

Comment:
Just curious about the timeout on this call. If it hangs, it seems like the struct will be locked. Could we lock, copy, unlock and then use the value ?

Reply:
we timeout after 2 seconds  `c.clcRunnerAPIClient.Timeout = 2 * time.Second`, wdyt?

Reply:
Yes, I think we should copy otherwise we spend 2 seconds in deadlock

Reply:
done